libkv.go

```diff
@@ -30,29 +33,98 @@ func NewLibKV(url *url.URL) *LibKV {
 
 	sourceType = ""
 	client = ""
+	options := store.Config{}
 
 	if url.Scheme == "consul" {
 		consul.Register()
 		sourceType = store.CONSUL
 		client = env.Getenv("CONSUL_HTTP_ADDR", "localhost:8500")
+		if timeout := env.Getenv("CONSUL_TIMEOUT", ""); timeout != "" {
+			num, err := strconv.ParseInt(timeout, 10, 16)
+			if err != nil {
+				logFatal("consul timeout error", err)
+			}
+			options.ConnectionTimeout = time.Duration(num) * time.Second
+		}
+		if ssl := env.Getenv("CONSUL_HTTP_SSL", ""); ssl != "" {
```

Comment:
CONSUL_HTTP_ADDR is the expected environment variable:

http://techblog.zeomega.com/devops/golang/2015/06/09/consul-kv-api-in-golang.html

The reason it talks about "local agent" is that the expected usage is you run an agent on every machine (and therefore always refer to it as localhost), but the full servers (1, 3 or 5) are the ones which actually store the data and are talked to via the agents.

The go API will automatically pick up that env variable, it is just libkv which overwrites it.

I think it would be more confusing for users of Consul to have all the normal env variables (e.g. CONSUL_HTTP_TOKEN  - as per https://www.consul.io/docs/commands/index.html#environment-variables) but then something totally different for the server address (a bit like we are using the same env variable for Vault as you would with the Vault CLI).